python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py

```diff
@@ -140,12 +200,15 @@ def run_model(
         :param outputs: outputs of the model that should be predicted
         :param selected_atoms: subset of atoms on which to run the calculation
         """
-        types, positions, cell = _ase_to_torch_data(atoms)
+        types, positions, cell = _ase_to_torch_data(
+            atoms=atoms, dtype=self.dtype, device=self.device
+        )
         system = System(types, positions, cell)
 
         # Compute the neighbors lists requested by the model using ASE NL
         for options in self._model.requested_neighbors_lists():
             neighbors = _compute_ase_neighbors(atoms, options)
+            neighbors = neighbors.to(dtype=self.dtype, device=self.device)
```

Comment:
Isn't `neighbors` an object with mixed ints and floats? What happens here, do the ints get turned into floats?

Reply:
Only the values (here the distance vectors) of the `TensorBlock` are moved to another dtype. The Labels are always kept as `torch.int32` (but can be copied to another device).